extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,972 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
+- `style` is the **state variable**. It corresponds to the matching JSON key, and sets the bound property to that key's value.
+
+[/filter] <!-- formats="email" -->
+
+### `<amp-state>` specification
+
+[filter formats="websites, stories, ads"]
+An `amp-state` element may contain either a child `<script>` element **OR** a `src` attribute containing a CORS URL to a remote JSON endpoint, but not both.
+
+```html
+<amp-state id="myLocalState">
+  <script type="application/json">
+    {
+      "foo": "bar"
+    }
+  </script>
+</amp-state>
+
+<amp-state id="myRemoteState" src="https://data.com/articles.json"> </amp-state>
 ```
 
 [/filter] <!-- formats="websites, stories, ads" -->
 
 [filter formats="email"]
+An `amp-state` element must contain a child `<script>` element.
 
 ```html
-<!-- Store complex nested JSON data in <amp-state> elements. -->
-<amp-state id="myAnimals">
+<amp-state id="myLocalState">
   <script type="application/json">
     {
-      "dog": {
-        "imageUrl": "/img/dog.jpg",
-        "style": "greenBackground"
-      },
-      "cat": {
-        "imageUrl": "/img/cat.jpg",
-        "style": "redBackground"
+      "foo": "bar"
+    }
+  </script>
+</amp-state>
+```
+
+[/filter] <!-- formats="email" -->
+
+[filter formats="websites, stories, ads"]
+
+#### Attributes
+
+[filter formats="websites, stories, ads"]
+
+##### `src` (optional)
+
+The URL of the remote endpoint that will return the JSON and update this `amp-state`. This must be a CORS HTTP service. The `src` attribute allows all standard URL variable substitutions. See the [Substitutions Guide](../../spec/amp-var-substitutions.md) for more info.
+
+[tip type="important"]
+The endpoint must implement the requirements specified in the [CORS Requests in AMP](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cors-requests) spec.
+[/tip]
+
+##### `credentials` (optional)
+
+Defines a `credentials` option as specified by the [Fetch API](https://fetch.spec.whatwg.org/).
+
+- Supported values: `omit`, `include`
+- Default: `omit`
+
+To send credentials, pass the value of `include`. If this value is set, the response must follow the [AMP CORS security guidelines](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cors-requests/#cors-security-in-amp).
+[/filter] <!-- formats="websites, stories, ads" -->
+
+[filter formats="email"]
+
+#### Actions
+
+##### `refresh`
+
+The `refresh` action refetches data from data point the `src` attribute points to. It ignores the browser cache.
+
+[example preview="inline" playground="true" imports="amp-bind"]
+
+```html
+<amp-state id="currentTime" src="/documentation/examples/api/time"></amp-state>
+<button on="tap:currentTime.refresh">
+  Refresh
+</button>
+<div [text]="currentTime.time"></div>
+```
+
+[/example]
+
+A common need for `refresh` is working with live content that needs to be updated, such as a live sports score.
```

Comment:
Might be worth a note that we think AMP Script is a better fit for live content, but `amp-bind` will work for a subset of cases.